lib/faraday/connection.rb

```diff
@@ -464,5 +442,14 @@ def set_authorization_header(header_type, *args)
         header(*args)
       headers[Faraday::Request::Authorization::KEY] = header
     end
+
+    def find_default_proxy
+      warn 'no_proxy is unsupported' if ENV['no_proxy'] || ENV['NO_PROXY']
```

Comment:
Yes, I agree it is a weird way to use it, but it is allowed.  The find_proxy method should be moved to when the request is created in `build_request`.